src/cli/common.rs

```diff
@@ -381,70 +379,26 @@ where
 }
 
 pub(super) fn list_items(
-    distributable: DistributableToolchain<'_>,
-    f: impl Fn(&ComponentStatus) -> Option<&str>,
+    items: impl Iterator<Item = (String, bool)>,
```

Comment:
How about a named `struct ListItem { name: String, installed: bool }` to make it clearer what `(String, bool)` actually means here?